test/functional/services/filter_bar.js

```diff
@@ -56,25 +58,40 @@ export function FilterBarProvider({ getService }) {
       await testSubjects.click(`filter & filter-key-${key} pinFilter-${key}`);
     }
 
-    async addFilter(field, operator, values) {
-      if (!Array.isArray(values)) {
-        values = [values];
-      }
+    /**
+     * Adds a filter to the filter bar.
+     *
+     * @param {string} field The name of the field the filter should be applied for.
+     * @param {string} operator A valid operator for that fields, e.g. "is one of", "is", "exists", etc.
+     * @param {string[]|string} values The remaining parameters are the values passed into the individual
+     *   value input fields, i.e. the third parameter into the first input field, the fourth into the second, etc.
+     *   Each value itself can be an array, in case you want to enter multiple values into one field (e.g. for "is one of"):
+     * @example
+     * // Add a plain single value
+     * filterBar.addFilter('country', 'is', 'NL');
+     * // Add an exists filter
+     * filterBar.addFilter('country', 'exists');
+     * // Add a range filter for a numeric field
+     * filterBar.addFilter('bytes', 'is between', '500', '1000');
+     * // Add a filter containing multiple values
+     * filterBar.addFilter('extension', 'is one of', ['jpg', 'png']);
```

Comment:
Should be `filterBar.addFilter('extension', 'is one of', 'jpg', 'png');` (without brackets)

Reply:
Or, wait... should it? Looking in the code I guess you're handling arrays

Reply:
... I don't think so.  The way I read it is the first parameter goes into the first input box, and so on.  "is one of" means both jpg and png should go into the first input box.

Reply:
Yeah in case you want to add multiple values in the same field, you need to specify an array for that field, in case you leave the brackets it would try to enter those values into different fields... Slowly I am thinking to believe, this method has a confusing API no matter what I does :D